### ISSUE TITLE
Tests don't run with selenium-webdriver/testing mocha wrapper

### ISSUE THREAD

**Issue by tourdedave** [mmotkina/mocha-parallel-tests#47]
The official JavaScript Selenium bindings have a helper library for writing mocha tests which take advantage of the ControlFlow promise helper that comes in the bindings ([link](http://seleniumhq.github.io/selenium/docs/api/javascript/module/selenium-webdriver/testing/index.html)).

When I try to run tests written with it using `mocha-parallel-tests` it doesn't find them.

```text
mocha-parallel-tests test/*.js
  0 passing (12ms)
```

[My example code](https://github.com/selenium-guidebook/code-examples-javascript) is in a private repo, but I'll add you to it so you can take a look.

**Comment by 1999** [comment]
Thank you for posting issue. From what I see `selenium-webdriver/testing` wraps original `mocha` hooks (it, beforeEach etc) with `makeAsyncTestFn`. I'm not sure that I can fix this issue, but can you provide a bit more into on what's `ControlFlow` is and why is it important to wrap original `mocha` hooks with selenium-webdriver?

**Comment by tourdedave** [comment]
There's documentation about `ControlFlow` [here](http://seleniumhq.github.io/selenium/docs/api/javascript/module/selenium-webdriver/lib/promise_exports_ControlFlow.html). As for the rationale behind why it's necessary for selenium-webdriver, I'm not sure. I tried just using the built-in mocha hooks for my tests and wasn't able to get my tests to run correctly. I'll give it another go though.

**Comment by 1999** [comment]
> The control flow will ensure tasks are executed in the ordered scheduled, starting each task only once those before it have completed.
Maybe it's my english so bad, but what does this all mean? Mocha runs every test case (`it` hook) only when the previous case has been completed.

Anyway I have an idea how I can fix this issue, will look at it tomorrow.

**Comment by tourdedave** [comment]
Great!

**Comment by 1999** [comment]
One more observation:

```javascript
var builder = new webdriver.Builder().usingServer(url)
this.driver = builder.build()
```

When this code is running in `beforeEach()`, it's sync and test execution doesn't get paused. But `test.beforeEach()` pauses test executing.

**Comment by 1999** [comment]
Preparing new release...

**Comment by 1999** [comment]
@tourdedave could you please try `mocha-parallel-tests@1.2.0-beta.1`?

**Comment by 1999** [comment]
afaik something is still wrong. reopening for further fix.

**Comment by 1999** [comment]
https://gist.github.com/mattrlong/d64d6f95520e3211e15776d9b2dd0237#file-gistfile1-txt-L9

**Comment by mattrlong** [comment]
Can you expand on what you believe is the outstanding issue on this ticket? 

I myself am seeing the following error which I am not when using mocha directly:
  1)  "before all" hook:
     Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.

Calling done() should not be required when using the selenium-webdriver wrapper due to it's inbuilt promise manager.

**Comment by 1999** [comment]
Eh... Could you provide failing test case for this? Currently there's an existing [test case](https://github.com/mmotkina/mocha-parallel-tests/tree/master/test/run-programmatically) with timeout() and it seems to work as expected.

**Comment by mattrlong** [comment]
Right, sorry for the delay in responding. I've created a POC outside our current test pack, and the timeout is working as expected. I do see a separate issue which I will log separately however.

**Comment by 1999** [comment]
:+1:
